cupy/_math/sumprod.py

```diff
@@ -507,7 +507,65 @@ def gradient(f, *varargs, axis=None, edge_order=1):
         return outvals
 
 
-# TODO(okuta): Implement ediff1d
+def ediff1d(arr, to_end=None, to_begin=None):
+    """
+    Calculates the difference between consecutive elements of an array.
+
+    Args:
+        arr (cupy.ndarray): Input array.
+        to_end (cupy.ndarray, optional): Numbers to append at the end
+            of the returend differences.
+        to_begin (cupy.ndarray, optional): Numbers to prepend at the
+            beginning of the returned differences.
+
+    Returns:
+        cupy.ndarray: New array consisting differences among succeeding
+        elements.
+
+    .. seealso:: :func:`numpy.ediff1d`
+    """
+    # to flattened array.
+    arr = cupy.asanyarray(arr).ravel()
```

Comment:
To avoid implicit host-to-device sync, CuPy has a design policy that it accepts only `cupy.ndarray` where NumPy takes an arraylike. Would you remove `cupy.asanyarray`, then add a type check for `arr` instead?

Reply:
Yes, sure! Added. Thanks for the concept.